File: src/main/java/com/github/masx200/biliClient/api/video/VideoGet.java
package com.github.masx200.biliClient.api.video;

import com.github.masx200.biliClient.BiliCall;
import com.github.masx200.biliClient.BiliRequest;
import com.github.masx200.biliClient.able.Gettable;
import com.github.masx200.biliClient.model.video.Video;

/**
 * 描述：video get
 *
 * @author lpc lpc@hll520.cn
 * @version 1.0  2021-02-08-19:07
 * @since 2021-02-08-19:07
 */
public class VideoGet implements Gettable<Video> {

    private final BiliRequest request;

    public VideoGet(BiliRequest request) {
        this.request = request;
    }

    @Override
    public Video get() {
        return BiliCall.doCall(request).toData(Video.class);
    }
}

File: src/main/kotlin/com/github/masx200/biliClient/api/video/VideoGet.kt
package com.github.masx200.biliClient.api.video

import com.github.masx200.biliClient.BiliCall.doCall
import com.github.masx200.biliClient.BiliRequest
import com.github.masx200.biliClient.able.Gettable
import com.github.masx200.biliClient.model.video.Video
import java.util.function.Consumer
import org.apache.http.client.methods.HttpRequestBase

/**
 * 描述：video get
 *
 * @author lpc lpc@hll520.cn
 * @version 1.0  2021-02-08-19:07
 * @since 2021-02-08-19:07
 */
class VideoGet // 带参数的构造函数
// 默认构造函数
@JvmOverloads constructor(
    private val request: BiliRequest,
    private val beforeRequest: Consumer<HttpRequestBase?>? = null
) : Gettable<Video> {
    override fun get(): Video {
        return doCall(request, beforeRequest).toData<Video>()
    }
}
